next.clayui.com/content/docs/foundations/composing.md

```diff
@@ -3,4 +3,146 @@ title: "Composing"
+
+Designers cannot get caught up in strict rules this blocks the creative mind and can prevent UI advances and interaction enhancements. The same goes for components, it has to allow developers to move forward with the standards made by designers and allow them to reuse components in the future.
+
+Dealing with this is complex and requires balancing component flexibility and specialization, getting to that balance is difficult and we can spend a lot of time polishing and refining components to achieve **"component supremacy"**.
+
+We are delivering Clay as a **Multi-Layered API library**: **low-level**, **high-level** and **pure html**. Each approach has its purposes of use:
+
+<div class="clay-ullist">
+
+- **low-level** - Flexibility and use of first level composition so you can customize and create high level components.
+- **high-level** - Speed ​​to build new features are usually specialized components and tend to cover only a few use cases, limiting their flexibility.
+
+</div>
+
+#### Low-level components
+
+They are a great way to create new behaviors or extend, utilizing compositing to allow you to compose with other Clay components and build new possibilities.
+
+In essence, low-level components are more flexible but can slightly increase the "work" you have to do to get specifications, it requires you to do more compositions but that doesn't mean it's bad or the components are incomplete. Usually they tend to be [controlled components](https://reactjs.org/docs/uncontrolled-components.html).
+
+Clay follows a namespace pattern for low-level layer components, components are always nested according to responsibilities, common example:
+
+<div class="clay-ullist">
+
+- ClayDropDown
+- ClayDropDown.Action
+- ClayDropDown.Item
+- ClayDropDown.ItemList
+- ClayDropDown.Search
+
+</div>
+
+```jsx
+<ClayDropDown
+	active={expand}
+	onActiveChange={setExpand}
+	trigger={<ClayButton>{"Low-level"}</ClayButton>}
+>
+	<ClayDropDown.ItemList>
+		<ClayDropDown.Item href="#1">Linkable</ClayDropDown.Item>
+		<ClayDropDown.Item>
+			<ClayCheckbox
+				checked
+				label="Checkbox"
+				onChange={() => {}}
+			/>
+		</ClayDropDown.Item>
+	</ClayDropDown.ItemList>
+</ClayDropDown>
+```
+
+#### High-level components
+
+They are friendlier in simple use cases and help develop quickly but are specialized with specific cases. For their essence they are built on top of low-level components adding rules that help to reach [Lexicon](https://lexicondesign.io) specifications.
+
+The peculiarity is that often high-level components follow "inheritance" rather than composition but this is not a rule that separates the two layers, it may happen to follow composition.
+
+Unlike low level components that tend to be controlled in many cases high level can be treated as [uncontrolled](https://reactjs.org/docs/uncontrolled-components.html).
```

Comment:
Not sure this is necessarily true. We also try to also use controlled components for high-level. We do some internal wiring in some of them, but we still use onChange and value props typically.

Reply:
Yes, it's not 100% true but usually a high level tends to go in that direction, usually. I think there are cases with exception... but I think I will remove this so as not to create confusion.